File: source_funcs/go_transcoder/DIFFERENCE_BETWEEN_HIGHEST_AND_LEAST_FREQUENCIES_IN_AN_ARRAY/DIFFERENCE_BETWEEN_HIGHEST_AND_LEAST_FREQUENCIES_IN_AN_ARRAY.go
package main

import "sort"

func min(x int, y int) int {
	if x < y {
		return x
	}
	return y
}

func max(x int, y int) int {
	if x > y {
		return x
	}
	return y
}

func f_gold(arr []int, n int) int {
	sort.Ints(arr)
	var count int = 0
	var max_count int = 0
	var min_count int = n
	for i := int(0); i < (n - 1); i++ {
		if arr[i] == arr[i+1] {
			count += 1
			continue
		} else {
			max_count = max(max_count, count)
			min_count = min(min_count, count)
			count = 0
		}
	}
	return max_count - min_count
}

File: rust_funcs/go_transcoder/DIFFERENCE_BETWEEN_HIGHEST_AND_LEAST_FREQUENCIES_IN_AN_ARRAY/DIFFERENCE_BETWEEN_HIGHEST_AND_LEAST_FREQUENCIES_IN_AN_ARRAY.rs
fn min(x: i32, y: i32) -> i32 {if x < y { x } else { y } }
fn max(x: i32, y: i32) -> i32 {if x > y { x } else { y }}
fn cmpfunc(a: &i32, b: &i32) -> std::cmp::Ordering {a.cmp(b)}
fn len(arr: &[i32]) -> usize {arr.len()}
fn sort(arr: &mut [i32]) {arr.sort_by(cmpfunc);}

use std::cmp;

fn f_gold(arr: &mut [i32], n: i32) -> i32 {
    arr.sort();
    let mut count = 0; 
    let mut max_count = 0; 
    let mut min_count = n;
    for i in 0..(n - 1) {
        if arr[i as usize] == arr[(i + 1) as usize] {
            count += 1;
            continue; 
        } else {
            max_count = cmp::max(max_count, count);
            min_count = cmp::min(min_count, count);
            count = 0;
        }
    }
    max_count - min_count
}
